### ISSUE TITLE
Sending a reply fails with NSCalendar error

### ISSUE THREAD

**Issue by nolanw** [intercom/intercom-ios#328]
## Version info
  - Intercom for iOS version: 4.1.3
  - iOS version: 11.1 (iPhone X simulator) (Xcode 9.1 beta 2 (9B46))

## Expected behavior

I can reply in a conversation.

## Actual behavior

The reply seems to appear briefly then disappears, and is not received by the other side of the conversation.

## Steps to reproduce

1. Call `Intercom.presentMessenger()`.
1. Start a conversation by typing a message and pressing "Send".
2. Have other side reply twice: once with just words, then a second time with words and an attached image (it's possible there's a smaller reproduction here, haven't tested extensively).
3. Type a reply and press "send".

## Logs

Console:

```
*** -[NSCalendar isDate:inSameDayAsDate:]: date2 cannot be nil
Future exception.
A few of these errors are going to be reported with this complaint, then further violations will simply be ignored.
Here is the backtrace where this occurred this time (some frames may be missing due to compiler optimizations):
(
	0   CoreFoundation                      0x000000011674e4ac -[NSCalendar isDate:inSameDayAsDate:] + 284
	1   CoolApp                             0x000000010ec7b114 -[ICMTimestampCalculator part:isFromDifferentDayAsPart:] + 191
	2   CoolApp                             0x000000010ece8273 -[ICMNativeConversationContentViewController shouldAddDividerBeforePartAtIndex:inArray:] + 415
	3   CoolApp                             0x000000010ec32a2a -[Intercom_IntercomConversationCollectionViewController shouldAddDividerBeforeSection:] + 153
	4   CoolApp                             0x000000010ec336ad -[Intercom_IntercomConversationCollectionViewLayout createAttributesForSection:startY:] + 209
	5   CoolApp                             0x000000010ec33525 -[Intercom_IntercomConversationCollectionViewLayout prepareLayout] + 803
	6   UIKit                               0x00000001135d1837 -[UICollectionViewData _prepareToLoadData] + 156
	7   UIKit                               0x0000000113578062 -[UICollectionView _endItemAnimationsWithInvalidationContext:tentativelyForReordering:animator:] + 12822
	8   UIKit                               0x0000000113581805 -[UICollectionView _endUpdatesWithInvalidationContext:tentativelyForReordering:animator:] + 71
	9   UIKit                               0x0000000113581b4a -[UICollectionView _performBatchUpdates:completion:invalidationContext:tentativelyForReordering:animator:] + 435
	10  UIKit                               0x0000000113581974 -[UICollectionView _performBatchUpdates:completion:invalidationContext:tentativelyForReordering:] + 91
	11  UIKit                               0x00000001135818f6 -[UICollectionView _performBatchUpdates:completion:invalidationContext:] + 74
	12  UIKit                               0x000000011358184b -[UICollectionView performBatchUpdates:completion:] + 53
	13  CoolApp                             0x000000010ec2c97e -[Intercom_IntercomConversationCollectionViewController performBatchUpdates:completion:] + 375
	14  CoolApp                             0x000000010ec2e6f7 -[Intercom_IntercomConversationCollectionViewController setConversationParts:completion:] + 2258
	15  CoolApp                             0x000000010ec2ec3e -[Intercom_IntercomConversationCollectionViewController removeConversationPart:] + 243
	16  CoolApp                             0x000000010ece9526 -[ICMNativeConversationContentViewController removeAllTypingParts:] + 707
	17  Foundation                          0x00000001112cc1ee __NSFireTimer + 83
	18  CoreFoundation                      0x00000001166eb2a4 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
	19  CoreFoundation                      0x00000001166eaf62 __CFRunLoopDoTimer + 1026
	20  CoreFoundation                      0x00000001166eab1a __CFRunLoopDoTimers + 266
	21  CoreFoundation                      0x00000001166e2534 __CFRunLoopRun + 2308
	22  CoreFoundation                      0x00000001166e19b9 CFRunLoopRunSpecific + 409
	23  GraphicsServices                    0x0000000119e799c6 GSEventRunModal + 62
	24  UIKit                               0x0000000112a94948 UIApplicationMain + 159
	25  CoolApp                             0x000000010eb8a5ce main + 894
	26  libdyld.dylib                       0x000000011787bd81 start + 1
	27  ???                                 0x0000000000000001 0x0 + 1
)
```

**Comment by jtreanor** [comment]
Thanks for reporting this @nolanw. We look into it and get back to you when we know more.

**Comment by nolanw** [comment]
@bboyle18 Thought I'd check in on this issue and noticed the label change. Sounds like good news?

**Comment by bboyle18** [comment]
Hi @nolanw 👋 
Yes, we've added a change to prevent this crash in the future. It'll be available in the next release later this week.

Brian

**Comment by nolanw** [comment]
Great, thank you!

**Comment by bboyle18** [comment]
Fixed in [Release 4.1.4](https://github.com/intercom/intercom-ios/releases/tag/4.1.4)
